submissions/aymkin/OOP-JS (tjsw)/index.js

```diff
@@ -0,0 +1,96 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+
+Code repository: https://github.com/aymkin/a-tiny-JS-world
+Web app: https://aymkin.github.io/a-tiny-JS-world/
+   */
+
+function Inhobitant(species, name, gender, say, friends = []) {
+	this.species = species;
+	this.name = name;
+	this.gender = gender;
+	this.friends = friends;
+	this.say = say;
+}
+
+Inhobitant.prototype.makeSound = function() {
+	return this.say;
+};
+
+Inhobitant.prototype.getFriend = function() {
+	return this.hasOwnProperty('friends') && this.friends.length > 0
+		? `I have friends: ${this.friends.join(', ')}.`
+		: `I have no friends`;
+};
+
+Inhobitant.prototype.getBio = function() {
+	return `I am ${this.species}, my name is <strong>${
+		this.name
+	}</strong>. ${this.getLimbs()}. ${this.getFriend()} \nEvery day I say: ${this.makeSound()}.`;
+};
+
+Inhobitant.prototype.getLimbs = function() {
+	let limbs = 'The limbs are not defined on my constructor';
+	if (this.hasOwnProperty('legs')) {
+		limbs = `I have ${this.legs} legs`;
+	}
+	if (this.hasOwnProperty('hands')) {
+		limbs += ` and ${this.hands} hands`;
+	}
+	return limbs;
+};
+
+function Animal(species, name, gender, say, friends) {
+	Inhobitant.call(this, species, name, gender, say, friends);
+	this.legs = 4;
+}
+
+Animal.prototype = Object.create(Inhobitant.prototype);
```

Comment:
Animal has `legs` and those should have been printed with `.getBio` method.
This means that `Animal` has to define it's own `.getBio`, where it will call parent class `.getBio` and supplement the output of parent's `.getBio` with info re legs.